File: 프로그래머스/lv2/64065. 튜플/튜플.java
import java.util.*;
class Solution {
    public int[] solution(String s) {

        //문자열 앞뒤 {{ }} 삭제
        s = s.substring(2);
        s = s.substring(0,s.length()-2);
        
        List<String> arr = new ArrayList<>();
        
        String[] str = s.split("\\},\\{");
        
        //문자열 길이 만큼 정렬해줌
        Arrays.sort(str, Comparator.comparing(String::length));
        
        for(String e:str){
            String[] k = e.split(",");
            // System.out.println(Arrays.toString(k));
            for(int i=0;i<k.length;i++){
                if(arr.contains(k[i])==false){
                    arr.add(k[i]);
                }
            }
        }
        
        int [] A = new int[arr.size()];
        for(int i=0; i<arr.size();i++){
            A[i] = Integer.parseInt(arr.get(i));
        }
        

        
        return A;
    }
}

File: 프로그래머스/lv2/64065. 튜플/튜플.py
def solution(s):
    answer = []
    
    s=s[2:-2].split("},{")
    
    s = sorted(s, key=lambda x: len(x))
    
    for item in s:
        item = list(map(int, item.split(",")))
        for i in item:
            if i not in answer:
                answer.append(i)
    return answer